workers/social/lib/social/models/group/permissionset.test.coffee

```diff
@@ -0,0 +1,119 @@
+              'open group', { failure: -> done() }) client
+
+        it 'should allow for members', (done) ->
+          withConvertedUser ({ client }) ->
+            (JPermissionSet.permit \
+              'open group', { success: -> done() }) client
+
+
+      describe 'should support advanced permissions', ->
+
+        it 'should deny for guests', (done) ->
+          withDummyClient ({ client }) ->
+            (JPermissionSet.permit {
+              advanced : [ { permission: 'open group' } ]
+              failure  : -> done()
+            }) client
+
+        it 'should allow for members', (done) ->
+          withConvertedUser ({ client }) ->
+            (JPermissionSet.permit {
+              advanced : [ { permission: 'open group' } ]
+              success  : -> done()
+            }) client
+
+
+      describe 'should support multiple permissions', ->
+
+        it 'should deny for guests', (done) ->
+          withDummyClient ({ client }) ->
+            (JPermissionSet.permit {
+              advanced : [
+                { permission: 'grant access' }
+                { permission: 'open group' }
+              ]
+              failure  : -> done()
+            }) client
+
+        it 'should allow for members', (done) ->
+          withConvertedUser ({ client }) ->
+            (JPermissionSet.permit {
+              advanced : [
+                { permission: 'grant access' }
+                { permission: 'open group' }
+              ]
+              success  : -> done()
+            }) client
+
+
+      describe 'should support validators', ->
+
+        it 'should deny for guests when validator returned no', (done) ->
+          withDummyClient ({ client }) ->
+            (JPermissionSet.permit {
+              advanced : [
+                { permission: 'open group', validateWith: (..., cb) -> cb null, no }
+              ]
+              failure  : -> done()
+            }) client
+
+        it 'should allow for guests when validator returned yes', (done) ->
+          withDummyClient ({ client }) ->
+            (JPermissionSet.permit {
+              advanced : [
+                { permission: 'open group', validateWith: (..., cb) -> cb null, yes }
+              ]
+              success  : -> done()
+            }) client
+
+        it 'should deny for members when validator returned no', (done) ->
+          withConvertedUser ({ client }) ->
+            (JPermissionSet.permit {
+              advanced : [
+                { permission: 'open group', validateWith: (..., cb) -> cb null, no }
+              ]
+              failure  : -> done()
+            }) client
+
+        it 'should allow for guests when validator returned yes', (done) ->
+          withConvertedUser ({ client }) ->
+            (JPermissionSet.permit {
+              advanced : [
+                { permission: 'open group', validateWith: (..., cb) -> cb null, yes }
+              ]
+              success  : -> done()
+            }) client
+
```

Comment:
We might need to call `done` for the cases where we think it's gonna fail for each test, because, as far as i can see, it's gonna hang if the test isn't passing and gonna cause timeout. Not sure if it's intended but i guess if we call done with `err` it should fail early without waiting a timeout.